src/workflows/trades.ts

```diff
@@ -20,16 +20,13 @@ export async function createTradeWorkflow(
     },
   });
 
-  const { signable_message: signableMessage, payload_hash: payloadHash } =
-    signableResult.data;
+  const { signable_message: signableMessage } = signableResult.data;
 
   // Sign message with L1 credentials
-  const ethSignature = await signRaw(signableMessage, signer);
+  const ethSignature = await signRaw(signableMessage, l1Signer);
 
   // Sign hash with L2 credentials
-  const starkSignature = serializeSignature(
-    sign(starkWallet.starkKeyPair, payloadHash),
-  );
+  const starkSignature = await l2Signer.signMessage(l2Message);
```

Comment:
Why does this not sign the payloadHash?

Reply:
@zatlite you are right, should sign with the **payloadHash**.

Reply:
Make sense